ONSdigital/census-rm-sample-loader
# Generate and validate a sample with SPG cases

# Motivation and Context
Need to be able to generate and validate a sample file that contains SPG cases.

# What has changed
- Refactored the code that was used for CE cases, as SPG cases will be generated in a similar way - by having an E case then 3 U cases against 1 in 11 of the E cases.  
- Replaced `community_level` and `community_establishment` with `address_level` and `address_type` to allow for SPGs.  
- Added 2 SPG treatment codes (the 2 that we know about so far) to the validator and treatment_codes file.  If these change then it'll be a small change to update these, which would need to be done anyway.

# How to test?
- All unit tests should pass
- Generate a sample file.  Feel free to lower the numbers the `treatment_code_quantities.csv` so it doesn't take so long --> `pipenv run python generate_sample_file.py`.  Check that SPG cases are created in the same format as the others and with the same sort of splits as CE cases.
- Validate a sample file containing SPG cases --> `pipenv run python validate_sample.py <sample_file>`.  All should be good.

# Links
https://trello.com/c/1IB69qys
